src/module/chat-message/document.ts

```diff
@@ -97,13 +97,12 @@ class ChatMessagePF2e extends ChatMessage<ActorPF2e> {
 
     /** Get the owned item associated with this chat message */
     get item(): Embedded<ItemPF2e> | null {
```

Comment:
get item() is used by modules and should always return a result. If you want to not embed spells in chat messages, you'll need to make this async. I don't quite mind if items that aren't in the actor are embedded, but it can work either way so long as this method continues to work.

Reply:
Basically, if we make this async, an update can be made to automated animations to fix their stuff hopefully. But if this returns null for async stuff, then we can't fix their module afterwards.

Continuing to store the spell source data in flags completely avoids any updates, but I know you wanna reduce the update burden.